homeassistant/components/home_connect/sensor.py

```diff
@@ -1,29 +1,95 @@
 """Provides a sensor for Home Connect."""
 
+from dataclasses import dataclass, field
 from datetime import datetime, timedelta
 import logging
 from typing import cast
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ENTITIES
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 import homeassistant.util.dt as dt_util
 
+from .api import ConfigEntryAuth, HomeConnectDevice
 from .const import (
+    ATTR_DEVICE,
     ATTR_VALUE,
+    BSH_EVENT_PRESENT_STATE_OFF,
     BSH_OPERATION_STATE,
     BSH_OPERATION_STATE_FINISHED,
     BSH_OPERATION_STATE_PAUSE,
     BSH_OPERATION_STATE_RUN,
+    COFFEE_EVENT_BEAN_CONTAINER_EMPTY,
+    COFFEE_EVENT_DRIP_TRAY_FULL,
+    COFFEE_EVENT_WATER_TANK_EMPTY,
     DOMAIN,
+    REFRIGERATION_EVENT_DOOR_ALARM_FREEZER,
+    REFRIGERATION_EVENT_DOOR_ALARM_REFRIGERATOR,
+    REFRIGERATION_EVENT_TEMP_ALARM_FREEZER,
 )
 from .entity import HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class HomeConnectSensorEntityDescription(SensorEntityDescription):
```

Comment:
I would imagine that this is only for the alarm entities, and the class name now implies its for every sensor

Reply:
Yeah, my goal was to get a single [description](https://github.com/beastie29a/core/pull/1/files#diff-04bb25b821c929e1f88647c802d692fdb154cc832f5570590c7c2f48feecf442R83-R96) and [entity parent class](https://github.com/beastie29a/core/pull/1/files#diff-04bb25b821c929e1f88647c802d692fdb154cc832f5570590c7c2f48feecf442R360-R364) for each sensor. Though because of the current implementation that does not utilize `entity_descriptions`, I didn't want to make any breaking changes before migrating entities over.